C/include/c4Base.h

```diff
@@ -142,6 +142,8 @@ void* c4base_retain(void *obj) C4API;
 void c4base_release(void *obj) C4API;
 
 // These types are reference counted and have c4xxx_retain / c4xxx_release functions:
+// NOTE that in debug builds retain and release cann throw exceptions despite being
+// marked noexcept.  Release builds are exception free.
```

Comment:
They can't, actually. The effect of `noexcept` is that, rather than allow an exception to be thrown, the C++ runtime will terminate the process. This basically turns exceptions into fatal assertion failures (which our terminate handler logs.)

Reply:
That is predicated on calling the C function that actually wires it up, which I don't think any of the platforms do unfortunately.  I'd like to see that be automatic....

Reply:
I modified the comments